homeassistant/components/mqtt/models.py

```diff
@@ -91,17 +93,21 @@ def _convert_outgoing_payload(
 
             return payload
 
-        if self._attr_command_template is None:
+        if self._command_template is None:
             return value
 
-        values = {"value": value}
+        values: dict[str, Any] = {"value": value}
         if self._entity:
             values[ATTR_ENTITY_ID] = self._entity.entity_id
             values[ATTR_NAME] = self._entity.name
+            if self._command_template.hass:
+                values[ATTR_THIS] = template.TemplateStateFromEntityId(
```

Comment:
We don't need to create a `TemplateStateFromEntityId` object each time the template is rendered. Do it the first time only instead.

Reply:
Used setdefault now